### PR TITLE
PYTHON-2883 Regex decoding error tests in top.json have unexpected, invalid syntax

### PR DESCRIPTION
**Pull request by juliusgeo**
https://jira.mongodb.org/browse/PYTHON-2883

### CODE REVIEW

File: bson/json_util.py [hunk 1]
```diff
@@ -634,6 +634,8 @@ def _parse_canonical_code(doc):
 def _parse_canonical_regex(doc):
     """Decode a JSON regex to bson.regex.Regex."""
     regex = doc['$regularExpression']
+    if not isinstance(regex['options'], str):
+        raise TypeError('Bad $regularExpression options, options must be string')
```
Comment (ShaneHarvey) [inline bson/json_util.py:638]: Please move this check to below the len(doc) and len(regex) checks to ensure we give good error messages in all cases. For the same reason can you also add the type() of regex['options'] to the error message?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bson/json_util.py (hunk 1 above)
Comment (juliusgeo) [reply bson/json_util.py]: Done.

File: bson/json_util.py [hunk 2]
```diff
@@ -639,6 +639,8 @@ def _parse_canonical_regex(doc):
     if len(regex) != 2:
         raise TypeError('Bad $regularExpression must include only "pattern"'
                         'and "options" components: %s' % (doc,))
+    if not isinstance(regex['options'], str):
+        raise TypeError('Bad $regularExpression options, options must be string, was type %s' % (type(regex['options'])))
```
Comment (ShaneHarvey) [inline bson/json_util.py:643]: We limit our lines to 79 chars according to pep 8: https://www.python.org/dev/peps/pep-0008/#maximum-line-length

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bson/json_util.py (hunk 2 above)
Comment (juliusgeo) [reply bson/json_util.py]: Done.

File: bson/json_util.py [hunk 3]
```diff
@@ -639,6 +639,9 @@ def _parse_canonical_regex(doc):
     if len(regex) != 2:
         raise TypeError('Bad $regularExpression must include only "pattern"'
                         'and "options" components: %s' % (doc,))
+    if not isinstance(regex['options'], str):
+        raise TypeError('Bad $regularExpression options, options must be '
+                        'string, was type %s' % (type(regex['options'])))
     return Regex(regex['pattern'], regex['options'])
```
Comment (ShaneHarvey) [inline bson/json_util.py:645]: I suggest refactoring to `regex['options']` to a local variable to avoid looking it up repeatedly and improve readability.

**Review by ShaneHarvey — COMMENTED** [review]
One minor comment

File: bson/json_util.py (hunk 3 above)
Comment (juliusgeo) [reply bson/json_util.py]: Done.

**Review by ShaneHarvey — APPROVED** [review]
LGTM

**Pull request merged**